### PR TITLE
Introduce in-repo proposal process

### PR DESCRIPTION
**Pull request by KnVerey**
As discussed at recent SIG meetings, we're introducing a lighter weight process for Kustomize features that are only of interest to the Kustomize subproject and do not require the full Enhancements tracking process. Here's a first take on what that could look like. Detailed feedback greatly appreciated.

For ease of reference:
- [Normal KEP template](https://github.com/kubernetes/enhancements/blob/master/keps/NNNN-kep-template/README.md#user-stories-optional)
- [Proposal template for Cluster API](https://github.com/kubernetes-sigs/cluster-api/blob/master/docs/proposals/YYYYMMDD-template.md)
- [Proposal template for Gateway API](https://github.com/kubernetes-sigs/gateway-api/blob/master/site-src/geps/gep-696.md)

/cc @monopole @natasha41575

### CODE REVIEW

File: proposals/00-00-template.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+  At minimum, you should fill in the "Summary" and "Motivation" sections.
+- [ ] **Create a PR.**
+  Ping `@kubernetes-sigs/kustomize-admins` and `@kubernetes-sigs/kustomize-maintainers`.
+-->
+
+# Your short, descriptive title
+
+**Authors**:
+
+**Reviewers**:
```
Comment (yuwenma) [inline proposals/00-00-template.md:18]: Is the reviewers same as approvers? Can we give some doc to mention who can be a qualified reviewer (like kustomize members)?

Comment (KnVerey) [reply proposals/00-00-template.md:18]: Here's the doc on the Kustomize contributor ladder: https://github.com/kubernetes-sigs/kustomize/blob/master/CONTRIBUTING.md#contributor-ladder. I'd say anyone on either list (admin or maintainer) can be a reviewer here.

Comment (yuwenma) [reply proposals/00-00-template.md:18]: Gotcha. Thanks!

File: proposals/00-00-template.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+1. A non-goal
+1. Another non-goal
+
+## Proposal
+
+Proof of concept PR (optional):
+
+<!--
+This is where we get down to the specifics of what the proposal actually is. What is the plan for implementing this feature? Does this proposal require new kinds, fields or flags? What do they look like?
+-->
```
[the remark this answers is not in the document]
Comment (KnVerey) [reply proposals/00-00-template.md:70]: I'll add something like `Will this feature require extending the public interface of Kustomize's Go packages? (it's ok if you're not sure yet)` and try to emphasize in general that we don't want too much detail. I don't think most non-maintainers will know the answer re: packages. If we think the implementation will be problematic for a given feature, we'll probably need to be the ones to call it out in the proposal review.

**Comment by KnVerey** [comment]
Updated

**Comment by natasha41575** [comment]
/lgtm

It seems that we have a consensus that overall this looks good, and all the feedback so far has been incorporated. I think we can make it available for use now and iterate/improve this if we discover any friction when people try to use it. Thanks for working on this!

**Pull request merged**